test/python/test_rticonnextdds_data_access.py

```diff
@@ -319,3 +319,44 @@ def test_reset_sequence(self, test_output, test_input):
     # The other fields are unchanged:
     assert sample.get_number("my_point.x") == 3
     assert sample.get_number("my_point_sequence#") == 2
+
+  def test_get_dictionary(self, populated_input):
+    # populated_input[0] contains test_dictionary
+    sample = populated_input[0]
+
+    # Attempt to get_dictionary for non existent member
+    with pytest.raises(rti.DdsError, match=r".*Cannot find.*IDoNotExist.*") as excinfo:
+      sample.get_dictionary("IDoNotExist")
+
+    # Attempt to get_dictionary for non-complex members
+    with pytest.raises(rti.DdsError, match=r".*invalid TypeCode kind DDS_TK_LONG.*") as excinfo:
+      sample.get_dictionary("my_long")
+    with pytest.raises(rti.DdsError, match=r".*invalid TypeCode kind DDS_TK_DOUBLE.*") as excinfo:
+      sample.get_dictionary("my_double")
+    with pytest.raises(rti.DdsError, match=r".*invalid TypeCode kind DDS_TK_BOOLEAN.*") as excinfo:
+      sample.get_dictionary("my_optional_bool")
+    with pytest.raises(rti.DdsError, match=r".*invalid TypeCode kind DDS_TK_LONG.*") as excinfo:
+      sample.get_dictionary("my_optional_long")
+    with pytest.raises(rti.DdsError, match=r".*invalid TypeCode kind DDS_TK_STRING.*") as excinfo:
+      sample.get_dictionary("my_string")
+    with pytest.raises(rti.DdsError, match=r".*invalid TypeCode kind DDS_TK_ENUM.*") as excinfo:
+      sample.get_dictionary("my_enum")
+    # It is possible to use get_dictionary to access nested members, but the nested
+    # member must be a complex type
+    with pytest.raises(rti.DdsError, match=r".*invalid TypeCode kind DDS_TK_LONG.*") as excinfo:
+      sample.get_dictionary("my_point.x")
+
+    # Valid values for member_name
+    the_point = sample.get_dictionary("my_point")
+    assert the_point['x'] == 3 and the_point['y'] == 4
```

Comment:
Even though it is against what we would do in the core, I think we should allow concatenating asserts like this in python. It is obvious which one has failed in the resultant message (e.g., `assert failed 3 == 3 and 1 == 4`)